### ISSUE TITLE
checksumChanges.filter is not a function

### ISSUE THREAD

**Issue by trashcoder** [xemle/home-gallery#13]
After a graceful shutdown via ctrl-c i get this error message:

Graceful shutdown. Ids of 13.1GB (3.2%) were calculated, 4.3% of 415.8GB are done. Please be patient to avoid data loss!
cli.index.update error Failed to create index: TypeError: checksumChanges.filter is not a function

**Comment by xemle** [comment]
Oh. Yes. I found the reason. Thank you for pointing out and filing the bug. Have not done this step for a long time.

The index should be fine and the error happens after a the safety index write in the next step creating the index journal. So you can continue testing the gallery.

Steps are:

* Read index
* Calculating missing checksums
--> `ctrc-c`
* Abort and write current index state
* Create index journal --> failure

I will provide a fix today or in the next days.


Side note: If you are using cifs: Are you aware using a good file change strategy via index matcher property in the [configuration](https://docs.home-gallery.org/configuration/index.html) like `size-ctime-inode` or `size-ctime`?

**Comment by xemle** [comment]
Another side note: Since you have a relatively large size to index I guess you are not using the incremental import as proposed in the [quickstarts](https://docs.home-gallery.org/install/index.html).

If true I recommend it at the first place. The files are imported and processed in chunks and become visible in the browser after a chunk is done. So you can test and verify your expectations to the new photo gallery software.

The chunks are based on file count. The import process might take longer if there are lots of videos in it.

**Comment by trashcoder** [comment]
> Side note: If you are using cifs: Are you aware using a good file change strategy via index matcher property in the [configuration](https://docs.home-gallery.org/configuration/index.html) like `size-ctime-inode` or `size-ctime`?
Ah, i see, a good idea. I'l test it today.

> Another side note: Since you have a relatively large size to index I guess you are not using the incremental import as proposed in the [quickstarts](https://docs.home-gallery.org/install/index.html).
good idea. i'l also try it today.

i stopped indexing and focused to test it on a subfolder with less files.

**Comment by xemle** [comment]
For fast testing feedback I recommend the initial (progressive) import with a only image exclude filter from the [repo](https://github.com/xemle/home-gallery/blob/master/examples/only-images.exclude). It is OK to run ist on the real large folder.

To start fresh it is enough to delete the file index. The gallery sees only these files which are indexed. Already calculated previewes are reused safely.

**Comment by trashcoder** [comment]
Yeah, but today i found out, that a disk attached to a fritz.box isn't a good NAS (1mb/s on USB3) . I'm actually building up a NAS within my proxmox environment for a better performance on my photo-library disk.

**Comment by xemle** [comment]
Since this bug does not break data I will include the fix in the next release
